include/swift/Runtime/Metadata.h

```diff
@@ -1760,7 +1758,14 @@ struct TargetFunctionTypeMetadata : public TargetMetadata<Runtime> {
   TargetPointer<Runtime, const Argument> getArguments() const {
     return reinterpret_cast<TargetPointer<Runtime, const Argument>>(this + 1);
   }
-  
+
+  uint32_t getParameterFlags(unsigned index) const {
+    auto numParams = getNumArguments();
+    assert(index < numParams);
+    auto *flags = reinterpret_cast<const uint32_t *>(this + 1 + numParams);
```

Comment:
This pointer computation seems very wrong; the ``+ numParams`` is scaled by sizeof(FunctionTypeMetadata).

Also, I'm not sure I understand why you're being incremental about changing the ABI here, at least as regards the layout of FunctionTypeMetadata.  Is there a good reason not to design and implement the desired ABI at this point, since you're revising the ABI anyway?  Why force someone else to come by and do all these same updates again?

Reply:
I will redo the pointer but I don’t understand what do you mean by incremental changes here, can you please elaborate? I am thinking of doing everything related to function metadata ABI, this is just a first changes wrt the layout I wanted input on...

Reply:
This seems to be implementing a flat, always-present array.  The things we talked about:
  - maybe not representing this as an array at all
  - maybe only including elements in the array if they have flags
  - or at least allowing the array to be optional

Every change you make here has to be quite involved in terms of e.g. updating all the runtime functions and the compiler's calls to them and how MetadataReader interprets the runtime structures, so I'm questioning whether it really makes sense to do all these changes in separate patches.

Reply:
Ah sure, I understand. I'm planning to do all this next as one step and update existing commit, just wanted to make sure that I haven't done something obviously terrible since I'm pretty new to LLVM...